src/cli/CommandsLogic.ts

```diff
@@ -294,26 +299,27 @@ export default class CommandsLogic {
       console.log(`== Saved RelayHub address at HubId:"${deployOptions.registryHubId}" to VersionRegistry`)
     }
 
-    let paymasterAddress = constants.ZERO_ADDRESS
+    let pmInstance: Contract | undefined
+    let paymasterVersion = ''
     if (deployOptions.deployPaymaster === true) {
-      const pmInstance = await this.deployPaymaster(Object.assign({}, options), rInstance.options.address, deployOptions.from, fInstance, deployOptions.skipConfirmation)
-      paymasterAddress = pmInstance.options.address
-
-      // Overriding saved configuration with newly deployed instances
-      this.config.paymasterAddress = paymasterAddress
+      pmInstance = await this.deployPaymaster(Object.assign({}, options), rInstance.options.address, deployOptions.from, fInstance, deployOptions.skipConfirmation)
+      paymasterVersion = await pmInstance.methods.versionPaymaster()
     }
-    this.config.relayHubAddress = rInstance.options.address
 
     await registerForwarderForGsn(fInstance, options)
 
-    return {
+    this.deployment = {
+      paymasterVersion,
       relayHubAddress: rInstance.options.address,
       stakeManagerAddress: sInstance.options.address,
       penalizerAddress: pInstance.options.address,
       forwarderAddress: fInstance.options.address,
       versionRegistryAddress: regInstance.options.address,
-      naivePaymasterAddress: paymasterAddress
+      paymasterAddress: pmInstance?.options.address ?? constants.ZERO_ADDRESS
```

Comment:
Ok, so there was an interface called 'DeploymentResult', and it could only be generated by the `gsn-deploy` command so it made sense for it to be called 'naivePaymaster'. Now I have turned this interface to a more widely used 'GSNContractsDeployment' so it stopped making sense to have two basically identical interfaces with one having a special name for a paymaster it deploys...